app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,136 @@
+class OrdersController < ApplicationController
+  
+  before_action :authenticate_user!
+  before_action :set_journey, only: %i(new create checkout)
+  before_action :set_order, only: %i(checkout charge)
+  before_action :instantiate_order, only: %i(create)
+  before_action :ensure_booking_is_active, only: %i(new checkout create)
+  
+  def checkout
+  end
+  
+  def new
+    @minutes = calculate_minutes
+    @order = Order.new
+    authorize(@order)
+    # add here all the sum of all the minutes that have not been paid. 
+  end 
+  
+  def create
+    @order.consumer_total = @order.minutes * @order.price_hour / 60
+    @order.state = 'not paid'
+    if @order.valid?
+      authorize(@order)
+      if @order.save
+        flash[:success] = t 'orders.create.success'
+        redirect_to dashboard_path
+      else
+        flash.now[:error] = t 'orders.create.unexpected_error'
+        handle_errors
+      end
+    else
+      flash.now[:error] = t 'orders.create.validation_error'
+      authorize(@order, :true?)
+      handle_errors
+    end
+  end
+  
+  def charge
+    
+    # stripe_token = params.require :stripeToken
+    # if @order.charge! stripe_token
+    #   flash[:success] = t '.success'
+    #   redirect_to @order.reservation_link
+    # else
+    #   @order.reload # for cleaning up any possibly modified attributes
+    #   flash[:error] = t 'stripe.payment_failed'
+    #   redirect_to @order.payment_link
+    # end
+    
+    customer = Stripe::Customer.create(
+      source: params[:stripeToken],
+      email:  params[:stripeEmail]
+    )
+
+    charge = Stripe::Charge.create(
+      customer:     customer.id,   
+      amount:       @order.consumer_total_cents, 
+      description:  @order.charge_description,
+      currency:     @order.consumer_total.currency
+    )
+    
+  @order.update(payment: charge.to_json, state: 'paid')
+  flash[:success] = t '.success'
+  redirect_to dashboard_path
+    
+  rescue Stripe::CardError => e
+    flash[:error] = e.message
+    redirect_to dashboard_path
+    
```

Comment:
I can only introduce valid cards in stripe. For instance when I test stripe with an expired card I get: 
```Render and/or redirect were called multiple times in this action. Please note that you may only call render OR redirect, and at most once per action. Also note that neither redirect nor render terminate execution of the action, so if you want to exit an action after redirecting, you need to do something like "redirect_to(...) and return". ```
This is because I am unsure on how to locate the `rescue` action. I could do it as you do (with `if`/`else`, but i would like to do it like this in order to get the `e` error from stripe. Any ideas on how I I could achieve this?